### PR TITLE
add customized config and stack files for community GSI/EnKF

### PR DESCRIPTION
**Pull request by guoqing-noaa**
Community GSI/EnKF users can use config_comgsi.sh and stack_comgsi.yaml to build only those libraries required to compile regional GSI/EnKF.

### CODE REVIEW

**Comment by edwardhartnett** [comment]
Do these packages have a cmake build system?

We need to get a cmake build system for all packages in hpc-stack. If there is not currently one, we need to come up with a plan to transition to a cmake build.

We cannot support home-rolled build scripts for 50+ libraries and packages, so we require a standard build system.

**Comment by kgerheiser** [comment]
@edwardhartnett this doesn't really have anything to do with a build system. It's the same packages that are already here, but with some disabled because GSI doesn't need them.

**Comment by edwardhartnett** [comment]
OK never mind...

File: config/config_comgsi.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+
+# Compiler/MPI combination
+export HPC_COMPILER=${HPC_COMPILER:-"intel/18.0.5"}
```
Comment (kgerheiser) [inline config/config_comgsi.sh:4]: What machine does this config refer to?

Comment (guoqing-noaa) [reply config/config_comgsi.sh:4]: That value is for Cheyenne but these three HPC_* variables are just place holders.  Community users needs to change them or set corresponding environments for hpc-stack to work (community users spread around the world, not limited to NOAA and NCAR machines). Thanks!

Comment (kgerheiser) [reply config/config_comgsi.sh:4]: Do we need this specific config then? I'm fine with the `stack_comgsi`, but I'm not sure this config adds anything.

Comment (guoqing-noaa) [reply config/config_comgsi.sh:4]: I would hope to add them. But I am okay if it is decided to not include them and if so, community users will need to get a copy from the comgsi/hpc-stack fork (not as official as from NOAA-EMC/hpc-stack).

Comment (kgerheiser) [reply config/config_comgsi.sh:4]: Maybe a different name then? It's not clear to me that this is meant to be used on Cheyenne, and none of the other settings are unique. You can't just run use this config on any machine. Or is GSI development usually done on Cheyenne?

Comment (guoqing-noaa) [reply config/config_comgsi.sh:4]: Do you have any suggestions on "a different name"?  Yes, I mainly develop and test community GSI on Cheyenne.

Comment (kgerheiser) [reply config/config_comgsi.sh:4]: How about `config_cheyenne_GSI?`

Comment (guoqing-noaa) [reply config/config_comgsi.sh:4]: Ah, I think that may not be appropriate. These two files are NOT used on Cheyenne solely.  I also used it on Orion to build libraries for regional GSI/EnKF ( I only need to export HPC_COMPILER and HPC_MPI variables). Users also use them to build their comgsi-flavor hpc-stack, for example on Stampede of TACC, on supercomputers at CSU, etc. I got what you mean. You'd like to see this file is specific to one particular machine. But my intention is to get ready a file specifically customized for building community GSI (mostly for regional applications). If this goal conflicts with some hpc-stack policies, it is fine to not include these two files. Thanks!

Comment (kgerheiser) [reply config/config_comgsi.sh:4]: I'm convinced with your explanation.

**Review by kgerheiser — APPROVED** [review]

File: config/config_comgsi.sh (hunk 1 above)
Comment (guoqing-noaa) [reply config/config_comgsi.sh:4]: Thanks!

**Comment by guoqing-noaa** [comment]
Could someone merge this PR as it was already approved by @kgerheiser ?  @Hang-Lei-NOAA

**Pull request merged**